stackdriver/config/config.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"sync"
+
+	"google.golang.org/api/option"
+
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+var (
+	mtx sync.RWMutex
+)
+
+const (
+	// ConfigMapNameEnv is the name of the environment variable that contains the name of
+	// the ConfigMap for stackdriver configuration.
+	ConfigMapNameEnv = "CONFIG_STACKDRIVER_NAME"
+	// secretNamespaceDefault is the namespace in which secrets for authenticating with Stackdriver
+	// should be stored.
+	secretNamespaceDefault = "default"
```

Comment:
Yes, I made it configurable because I realized that knative/eventing will not be in the "knative-serving" namespace. When I make changes to knative/serving, I will set it to "knative-serving" there.

Extra configuration will also be useful if users want to add the Secret in their own namespaces instead of "knative-serving"